src/core/lender/assessor.sol

```diff
@@ -112,4 +116,53 @@ contract Assessor is Math, DSNote, Auth {
     function _seniorDebt() internal returns (uint) {
         return (senior != address(0x0)) ? SeniorTrancheLike(senior).debt() : 0;
     }
-}
+
+
+    function calcMaxSeniorAssetValue() public returns (uint) {
+        uint juniorAssetValue = calcAssetValue(junior);
+        if (juniorAssetValue == 0) {
+            return 0;
+        }
+        // juniorAssetValue/minJuniorRatio * (ONE-minJuniorRatio) same as:
+        return sub(rdiv(juniorAssetValue, minJuniorRatio), juniorAssetValue);
+    }
+
+    function calcMinJuniorAssetValue() public returns (uint) {
+        uint seniorAssetValue = calcAssetValue(senior);
+        if (seniorAssetValue == 0) {
+            return uint(-1);
+        }
+        return rmul(rdiv(seniorAssetValue, ONE-minJuniorRatio), minJuniorRatio);
+    }
+
+    // only needed for external contracts
+    function currentJuniorRatio() public returns(uint) {
+        uint juniorAssetValue = calcAssetValue(junior);
+        return rmul(rdiv(ONE,(add(juniorAssetValue, calcAssetValue(senior)))), juniorAssetValue);
```

Comment:
maybe simplify like this: 
`rdiv(juniorAssetValue, add(juniorAssetValue, calcAssetValue(senior)));`